egui_for_winit/src/lib.rs

```diff
@@ -0,0 +1,638 @@
+                    // Treat as zoom instead:
+                    self.egui_input.zoom_delta *= (delta.y / 200.0).exp();
+                } else {
+                    self.egui_input.scroll_delta += delta;
+                }
+            }
+            // WindowEvent::TouchpadPressure {
+            //     device_id,
+            //     pressure,
+            //     stage,
+            //     ..
+            // } => {
+            //     // TODO
+            // }
+            WindowEvent::Touch(touch) => {
+                use std::hash::{Hash as _, Hasher as _};
+                let pixels_per_point_recip = 1. / self.pixels_per_point();
+                let mut hasher = std::collections::hash_map::DefaultHasher::new();
+                touch.device_id.hash(&mut hasher);
+                self.egui_input.events.push(egui::Event::Touch {
+                    device_id: egui::TouchDeviceId(hasher.finish()),
+                    id: egui::TouchId::from(touch.id),
+                    phase: match touch.phase {
+                        winit::event::TouchPhase::Started => egui::TouchPhase::Start,
+                        winit::event::TouchPhase::Moved => egui::TouchPhase::Move,
+                        winit::event::TouchPhase::Ended => egui::TouchPhase::End,
+                        winit::event::TouchPhase::Cancelled => egui::TouchPhase::Cancel,
+                    },
+                    pos: egui::pos2(
+                        touch.location.x as f32 * pixels_per_point_recip,
+                        touch.location.y as f32 * pixels_per_point_recip,
+                    ),
+                    force: match touch.force {
+                        Some(winit::event::Force::Normalized(force)) => force as f32,
+                        Some(winit::event::Force::Calibrated {
+                            force,
+                            max_possible_force,
+                            ..
+                        }) => (force / max_possible_force) as f32,
+                        None => 0_f32,
+                    },
+                });
+            }
+            WindowEvent::HoveredFile(path) => {
+                self.egui_input.hovered_files.push(egui::HoveredFile {
+                    path: Some(path.clone()),
+                    ..Default::default()
+                });
+            }
+            WindowEvent::HoveredFileCancelled => {
+                self.egui_input.hovered_files.clear();
+            }
+            WindowEvent::DroppedFile(path) => {
+                self.egui_input.hovered_files.clear();
+                self.egui_input.dropped_files.push(egui::DroppedFile {
+                    path: Some(path.clone()),
+                    ..Default::default()
+                });
+            }
+            _ => {
+                // dbg!(event);
+            }
+        }
+    }
+
+    /// Call with the output given by `egui`.
+    ///
+    /// This will update the cursor, copy text to the clipboard, etc.
```

Comment:
In some places it's better to work out 'etc'. It isn't clear what this might entail. Preferablly list all user notably actions